x-pack/platform/plugins/shared/lens/public/runtime_state/converters/raw_color_mappings/converter.ts

```diff
@@ -0,0 +1,164 @@
+        rules: [oldAssignment.rule],
+      };
+    }),
+  };
+}
+
+export function convertColorMappingAssignment(
+  oldAssignment: DeprecatedColorMappingConfig['assignments'][number],
+  column?: Partial<GenericIndexPatternColumn> | null
+): ColorMapping.Assignment {
+  return {
+    color: oldAssignment.color,
+    touched: oldAssignment.touched,
+    rules: convertColorMappingRule(oldAssignment.rule, column),
+  };
+}
+
+const NO_VALUE = Symbol('no-value');
+
+export function convertColorMappingRule(
+  rule: DeprecatedColorMappingConfig['assignments'][number]['rule'],
+  column?: Partial<GenericIndexPatternColumn> | null
+): ColorMapping.ColorRule[] {
+  switch (rule.type) {
+    case 'auto':
+      return [];
+    case 'matchExactly':
+      return rule.values.map((value) => {
+        const rawValue = convertToRawValue(value, column);
+
+        if (rawValue !== NO_VALUE) {
+          return {
+            type: 'raw',
+            value: rawValue,
+          };
+        }
+
+        return {
+          type: 'match',
+          pattern: String(value),
+          matchEntireWord: true,
+          matchCase: true,
+        };
+      });
+
+    // Rules below not yet used, adding conversions for completeness
+    case 'matchExactlyCI':
+      return rule.values.map((value) => ({
+        type: 'match',
+        pattern: Array.isArray(value) ? value.join(' ') : value,
+        matchEntireWord: true,
+        matchCase: false,
+      }));
+    case 'regex':
+      return [{ type: rule.type, pattern: rule.values }];
+    case 'range':
+    default:
+      return [rule];
+  }
+}
+
+const getParentFormatId = (column: Partial<GenericIndexPatternColumn>) =>
+  'params' in column
+    ? (column.params as { parentFormat?: { id?: string } })?.parentFormat?.id
+    : undefined;
+
+/**
+ * Attempts to convert the previously stringified raw values into their raw/serialized form
+ *
+ * Note: we use the `NO_VALUE` symbol to avoid collisions with falsy raw values
+ */
+export function convertToRawValue(
+  value: string | string[],
+  column?: Partial<GenericIndexPatternColumn> | null
+): SerializedValue | symbol {
+  if (!column) return NO_VALUE;
+
+  const { dataType } = column;
+  const type = getParentFormatId(column);
+
+  // all array values are multi-term
+  if (type === 'multi_terms' || Array.isArray(value)) {
+    if (typeof value === 'string') return NO_VALUE;
+    return new MultiFieldKey({ key: value }).serialize();
+  }
+
+  if (type === 'range') {
+    return RangeKey.isRangeKeyString(value) ? RangeKey.fromString(value).serialize() : NO_VALUE;
+  }
+
+  switch (dataType) {
+    case 'number':
+    case 'boolean':
+    case 'date':
+      if (value === '__other__') return value; // numbers can have __other__ as a string
+      const numberValue = Number(value);
+      if (isFinite(numberValue)) return numberValue;
+      break;
+    case 'string':
+    case 'ip':
+      return value; // unable to distinguish manually added values
```

Comment:
as per other comment discussion, it could be possible that the stringified value was actually and array of strings with a single string. I'm not sure if this have any impact here

Reply:
Same as above, is there a way to render data in Lens such that the value is an array of `ips`?

Reply:
Per discussion above, it appears not possible for this value to be an array.